datatypes/src/raster/operations/interpolation.rs

```diff
@@ -37,40 +39,21 @@ where
         let out_x_size = info_out.global_geo_transform.x_pixel_size();
         let out_y_size = info_out.global_geo_transform.y_pixel_size();
 
-        let parallelism = rayon::current_num_threads();
-        let rows_per_task =
-            num::integer::div_ceil(output.grid_array.shape.axis_size_y(), parallelism);
-
-        let chunk_size = output.grid_array.shape.axis_size_x() * rows_per_task;
+        let map_fn = |gidx: GridIdx2D, _current_pixel_value: Option<P>| {
+            let GridIdx([y, x]) = gidx;
+            let out_y_coord = out_upper_left.y + y as f64 * out_y_size;
```

Comment:
this is one (MAC/MAD) instruction for the CPU. I don't want to write and validate the same logic all the time to save that.

Reply:
but the logic was already there and I think the code was still quite readable (especially in comparison to the abominations we have elsewhere in code). Also, it's a lookup, an addition and a multiplication, so not quite one instruction. And it will be performed redundantly thousands of times. Also, I think it should be possible to encapsulate this to reduce redundant code, where you provide some code that is done for each row and then some code that is done for each cell. 

For now, I don't find this solution better than what was there previously because we have a redundant parameter in the map function and redundant computations.